lib/helpers/createSpecification.js

```diff
@@ -0,0 +1,45 @@
+/* eslint no-self-assign: 0 */
+
+/**
+ * Adds necessary properties for a given specification.
+ * @see https://goo.gl/Eoagtl
+ * @function
+ * @param {object} definition - The `definition` or `swaggerDefinition` from options.
+ * @returns {object} Object containing required properties of a given specification version.
+ */
+function createSpecification(definition) {
+  const specification = JSON.parse(JSON.stringify(definition));
+
+  // Properties corresponding to their specifications.
+  const v2 = [
+    'paths',
+    'definitions',
+    'responses',
+    'parameters',
+    'securityDefinitions',
+  ];
+  const v3 = [...v2, 'components'];
+
+  if (specification.openapi) {
+    specification.openapi = specification.openapi;
```

Comment:
What is the goal of this line ? Same question for the line 29 (`specification.swagger = specification.swagger;`) ?

Reply:
@alexisab thanks a lot for making reviews!

In overall, I want to refactor much of the codebase to avoid reassigning values and mutating objects, but at this point I decided to keep old ways of organizing properties to avoid going too far changing things which are not related to the open api specification support.

The idea behind these lines is that that, if the user has provided any of these 2 properties, swagger or openapi in the definition options, respect them. Otherwise the rest of the logic adds and removes properties ...